src/Symfony/Component/Translation/Provider/AbstractProvider.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translaton\Provider;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Translation\Provider\ProviderInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ */
+abstract class AbstractProvider implements ProviderInterface
+{
+    protected const HOST = 'localhost';
+
+    protected $client;
+    protected $logger;
+
+    protected $host;
+    protected $port;
+
+    public function __construct(HttpClientInterface $client = null, LoggerInterface $logger = null)
+    {
+        $this->client = $client;
+        if (null === $client) {
+            if (!class_exists(HttpClient::class)) {
```

Comment:
symfony/http-client is now a hard dependency of Translation component, so I don't see any case where `HttpClient` will not be instancied, but I could miss something?

Reply:
It should be a hard dependency of the provider, not the translation component, IIRC (I am currently on a phone) this is what it is to be in notifier, isn't it?

Reply:
You are right, symfony/http-client is a hard dependency of each Notifier bridge. We must do the same for Translation Providers :+1:

Reply:
yes an IMHO no need to have a hard dependency on the translation component itself